src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -287,6 +287,12 @@ protected virtual string GenerateUniqueId()
                         continue;
                     }
 
+                    if (attributeValueChunk is PreallocatedTagHelperAttributeChunk)
```

Comment:
Use `as` rather than `is` followed by an unconditional cast i.e. change the `RenderPreallocatedBoundAttribute()` parameter to be `PreallocatedTagHelperAttributeChunk`.